tycho-its/src/test/java/org/eclipse/tycho/test/feature/FeatureWithDependenciesTest.java

```diff
@@ -28,7 +29,11 @@ public void testFeatureRestriction() throws Exception {
 		Verifier verifier = getVerifier("feature.dependency", false, true);
 		verifier.executeGoals(List.of("clean", "package"));
 		verifier.verifyErrorFreeLog();
-		File pluginsFolder = new File(verifier.getBasedir(), "site/target/repository/plugins");
+		File pluginsFolder = new File(verifier.getBasedir(), "site/target/repository/plugins/");
 		assertTrue("No plugin folder created!", pluginsFolder.exists());
+
+        assertTrue("AopAlliance not found!", new File(verifier.getBasedir(), "site/target/repository/plugins/org.aopalliance_1.0.0.v201105210816.jar").exists());
+        assertFalse("Guava found!", new File(verifier.getBasedir(), "site/target/repository/plugins/com.google.guava_30.1.0.v20210127-2300.jar").exists());
```

Comment:
yes, i changed the test to reflect the old behaviour

Reply:
The old behavior was that guava is only included when you specify the exact version, so if you like to test old behavior then you must specify an exact version for guava and still assert it is there...

Reply:
My question would rather be why was it included with exact version.
To my understanding requirements should not be included at all. This is why at Xtext we list them separately in category.xml

Reply:
> My question would rather be why was it included with exact version.

Because it was inconsistent... no wait it was **like everyone want it** ...

> To my understanding requirements should not be included at all.

As explained in the ticket, for P2 these are *all* requirements, you can simply take a look at the p2 metadata of the feature and you will se that the only difference is the version range specification, so at the P2 level there is literally no difference.

> This is why at Xtext we list them separately in category.xml

And that's actually something that should not be required, as you already required them in your feature ...

Reply:
I guess this should be discussed in the ticket.
I just wanted to provide some code to discuss

Reply:
@cdietrich at least I think it should then cover the case:

1. exact version used
2. "0.0.0" version used
3. version range used

Reply:
I am not sure when I can work on this. My internet is more kbit ish so it takes eons to test things

Reply:
@cdietrich from my side, no need to hurry, as far as I can see this will require a fix in p2 to really fix it (if at all) and if there is a real issue one can simply use an older release.